### PR TITLE
New providers (non-english)

### PR DESCRIPTION
**Pull request by hexated**
- [AnimeSail](https://111.90.143.42) (Indonesian)
- [Tocanime](https://tocanime.co) (Vietnamese)
- [Yomovies](https://yomovies.plus) (Hindi)
- Acefile (extractor)
- SpeedoStream (extractor)

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return HomePageResponse(homePageList)
+    }
+
+    private fun Element.toSearchResult(): AnimeSearchResponse {
+        val title = this.selectFirst("h3 a")?.text()?.trim() ?: ""
+        val href = fixUrl(this.selectFirst("h3 a")!!.attr("href"))
+        val posterUrl = fixUrlNull(this.selectFirst("div.card-item-img")?.attr("data-original"))
+        val epNum = this.selectFirst("div.card-item-badget.rtl")?.text()?.let { num ->
+            if (num.contains("Preview")) {
+                num.filter { it.isDigit() }.toIntOrNull()?.minus(1)
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt:64]: move num.filter { it.isDigit() }.toIntOrNull() out of if statment

Comment (LagradOst) [reply app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt:64]: dry code is good code

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        return document.select("div.col-lg-3.col-md-4.col-6").map {
+            it.toSearchResult()
+        }
+
+    }
+
+    override suspend fun load(url: String): LoadResponse {
+        val document = app.get(url).document
+
+        val title = document.selectFirst("h1.title")?.text() ?: ""
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt:88]: 🤨 this should throw an error or return null

File: app/src/main/java/com/lagradost/cloudstream3/extractors/Acefile.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import com.lagradost.cloudstream3.utils.*
+
+class Acefile : ExtractorApi() {
+    override val name = "Acefile"
+    override val mainUrl = "https://acefile.co"
+    override val requiresReferer = false
+
+    override suspend fun getUrl(url: String, referer: String?): List<ExtractorLink> {
+        val sources = mutableListOf<ExtractorLink>()
+        with(app.get(url).document) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/Acefile.kt:14]: any reason to use with() when you can just use  app.get(url).document.select?

File: app/src/main/java/com/lagradost/cloudstream3/extractors/SpeedoStream.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.lagradost.cloudstream3.utils.M3u8Helper
+
+class SpeedoStream : ExtractorApi() {
+    override val name = "SpeedoStream"
+    override val mainUrl = "https://speedostream.com"
+    override val requiresReferer = true
+
+    override suspend fun getUrl(url: String, referer: String?): List<ExtractorLink> {
+        val sources = mutableListOf<ExtractorLink>()
+        with(app.get(url, referer = referer).document) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/SpeedoStream.kt:17]: with`?

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/YomoviesProvider.kt [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                    it.toSearchResult()
+                }
+                if (items.isNotEmpty()) homePageList.add(HomePageList(header, items))
+            }
+
+        return HomePageResponse(homePageList)
+    }
+
+    private fun Element.toSearchResult(): SearchResponse {
+        val title = this.selectFirst("h2")?.text()?.trim() ?: ""
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/YomoviesProvider.kt:41]: 🤨 this should return null

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/YomoviesProvider.kt [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        val href = fixUrl(this.selectFirst("a")?.attr("href").toString())
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/YomoviesProvider.kt:42]: this should return null if not found, not "null"

**Review by LagradOst — COMMENTED** [review]
Very minor code changes 👍

**Comment by hexated** [comment]
I think it's ready to merge now

**Pull request merged**